File: ios/Records.swift
import ExpoModulesCore

struct SessionConfig: Record {
  @Field var instructions: String?
  @Field var options: GenerationOptions?
  @Field var tools: [ToolConfig]?
  @Field var toolTimeout: Double?
  @Field var responseFormat: String?
  @Field var schema: [String: Any]?
}

struct GenerationOptions: Record {
  @Field var temperature: Double?
  @Field var maxTokens: Int?
  @Field var topK: Int?
}

struct ToolConfig: Record {
  @Field var name: String = ""
  @Field var description: String = ""
  @Field var parameters: [String: Any] = [:]
}

struct ToolCallEvent: Record {
  @Field var callId: String = ""
  @Field var toolName: String = ""
  @Field var arguments: [String: Any] = [:]
}

File: android/src/main/java/expo/modules/localllm/Records.kt
package expo.modules.localllm

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class SessionConfig : Record {
  @Field
  var instructions: String? = null

  @Field
  var options: GenerationOptionsRecord? = null
}

class GenerationOptionsRecord : Record {
  @Field
  var temperature: Float? = null

  @Field
  var maxTokens: Int? = null

  @Field
  var topK: Int? = null
}
